### PR TITLE
`v-chakra` directive

### LINKED ISSUE #230: [Feature]: Improve styling escape hatches for styles overide.
**Is your feature request related to a problem? Please describe.**
Currently in Chakra UI Vue, in order to apply a manual object of styles in the template as an escape hatch, you need to create the object in the template and and then bind the class to the element in the template.

This seems to work fine. However, after using this for a while, it seems quite verbose.

**Describe the solution you'd like**

I'd like to propose a better escape hatch, which would include using a `v-chakra` directive that accepts an object of styles and binds the class name to the element.

e.g.:
Basic usage 1: (Object of styles)
```vue
<div v-chakra="{ bg: 'tomato', color: 'white', p: 3 }" >I am happy div</div>
```

Advanced usage 2: (Function that accepts theme & color mode)
```vue
<template>
  <div v-chakra="useStyles" >I am happy div</div>
</template>

<script>
export default {
  computed: {
    useStyles: vm => (theme, colorMode) => { /* object of styles */ }
  }
}
</script>
```

**Pros:**
- Gives any basic HTML element the ability to be styled with using Chakra UI Vue.
- Can be extended to read attributes so as to bind styles via attributes. e.g.

```vue
<button v-chakra p="3" bg="blue.400" d="flex">
  Click Me
</button>
```
This allows non-chakra components to benefit from Chakra's constraint based design patterns.

**Additional context**
- Need to do some research around how this would work with Nuxt.js

### PR DESCRIPTION
**Pull request by codebender828**
# ⚡️ `v-chakra` directive 
This pull request implements the `v-chakra` directive implemented in #230
Closes #230 

## Description

**Base usage - No arguments**
```vue
<div v-chakra p="3" color="red.400" font-weight="bold">
  Welcome to Chakra directive
</div>
```
**With value - Styles object**
```vue
<div v-chakra="{
  p: 3,
  shadow: 'sm',
  h1: {
    bg: 'blue.100'
  },
}">
  <h1>Title</h1>
  <p>Text</p>
</div>
```
**With value - Function**
```vue
<div v-chakra="theme => ({
  shadow: 'sm',
  bg: theme.colors.blue[800],
  color: theme.colors.yellow[300],
  p: {
    fontWeight: 'bold',
    p: 3
  }
})">
  <p>Computed styles</p>
</div>
```

**Example Demo Button**
```vue
<button v-chakra="{
  ':hover': { bg: 'green.400' },
  ':focus': { shadow: 'outline' }
}" font-weight="bold" px="4" py="3" color="white" rounded="md" bg="blue.400" outline="none">
  Button
</button>
```
## TODO:
- [ ] Add tests
- [ ] Add SSR support

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by DominusKelvin** [comment]
Looking good

File: packages/chakra-ui-core/src/directives/chakra.directive.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }
+  const className = css(Css(pure)(theme))
+  return [className, pure]
+}
+
+/** Creates Chakra Directive */
+export default function createCharkaDirective (theme) {
+  return {
+    bind (el, binding, vnode) {
+      console.log(vnode)
```
Comment (koca) [inline packages/chakra-ui-core/src/directives/chakra.directive.js:39]: console.log ^^

File: packages/chakra-ui-core/src/directives/chakra.directive.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  if (hasBindingValue) {
+    const className = css(Css(styleObject)(theme))
+    return [className, pure]
+  }
+  const className = css(Css(pure)(theme))
+  return [className, pure]
+}
+
+/** Creates Chakra Directive */
+export default function createCharkaDirective (theme) {
```
Comment (koca) [inline packages/chakra-ui-core/src/directives/chakra.directive.js:36]: what about SSR?

Comment (codebender828) [reply packages/chakra-ui-core/src/directives/chakra.directive.js]: Just added SSR support in subsequent commit.

File: packages/chakra-ui-core/src/directives/chakra.directive.js (hunk 1 above)
Comment (codebender828) [reply packages/chakra-ui-core/src/directives/chakra.directive.js]: Haha. god catch. removed.

**Comment by codebender828** [comment]
@DominusKelvin @koca Yes, it's now SSR feature complete.

This docs build will fail because this PR bumps the version of `@chakra-ui/vue`. If you'd like to test the directive locally, checkout to `release/chakra-directive` and then run `yarn workspace @chakra-ui/nuxt dev`.

Click the "Go to directive" link to see the page with the directive consumption. You can also modify the `directive.vue` files.

**Comment by koca** [comment]
> @DominusKelvin @koca Yes, it's now SSR feature complete.
> 
> This docs build will fail because this PR bumps the version of `@chakra-ui/vue`. If you'd like to test the directive locally, checkout to `release/chakra-directive` and then run `yarn workspace @chakra-ui/nuxt dev`.
> 
> Click the "Go to directive" link to see the page with the directive consumption. You can also modify the `directive.vue` files.

Hey @codebender828, Fixed the tests 🎉  and @nuxtjs/emotion renders the css properly now :) 

You can see the ssr rendered css if you change the line 36 

```
expect(html).toContain('⚡️ Hello chakra-ui/vue')
```

to

```
expect(html).toContain('fail')
```

**Comment by codebender828** [comment]
Thank you @koca !

**Pull request merged**